keras_cv/models/backbones/densenet/densenet_backbone.py

```diff
@@ -0,0 +1,367 @@
+      name: string, block label.
+
+    Returns:
+      a function that takes an input Tensor representing an apply_conv_block.
+    """
+    if name is None:
+        name = f"conv_block_{backend.get_uid('conv_block')}"
+
+    x1 = x
+    x = layers.BatchNormalization(
+        axis=BN_AXIS, epsilon=BN_EPSILON, name=f"{name}_0_bn"
+    )(x)
+    x = layers.Activation("relu", name=f"{name}_0_relu")(x)
+    x = layers.Conv2D(
+        4 * growth_rate, 1, use_bias=False, name=f"{name}_1_conv"
+    )(x)
+    x = layers.BatchNormalization(
+        axis=BN_AXIS, epsilon=BN_EPSILON, name=f"{name}_1_bn"
+    )(x)
+    x = layers.Activation("relu", name=f"{name}_1_relu")(x)
+    x = layers.Conv2D(
+        growth_rate,
+        3,
+        padding="same",
+        use_bias=False,
+        name=f"{name}_2_conv",
+    )(x)
+    x = layers.Concatenate(axis=BN_AXIS, name=f"{name}_concat")([x1, x])
+    return x
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class DenseNetBackbone(Backbone):
+    """Instantiates the DenseNet architecture.
+
+    Args:
+        blocks: numbers of building blocks for the four dense layers.
+        include_rescaling: bool, whether to rescale the inputs. If set
+            to `True`, inputs will be passed through a `Rescaling(1/255.0)`
+            layer.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        input_tensor: optional Keras tensor (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+
+    Examples:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Pretrained backbone
+    model = keras_cv.models.DenseNetBackbone.from_preset("densenet121_imagenet")
+    output = model(input_data)
+
+    # Randomly initialized backbone with a custom config
+    model = DenseNetBackbone(
+        blocks=[6, 12, 24, 16],
+        include_rescaling=False,
+    )
+    output = model(input_data)
+    ```
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        *,
+        blocks,
+        include_rescaling,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        pooling=None,
+        **kwargs,
+    ):
+        inputs = utils.parse_model_inputs(input_shape, input_tensor)
+
+        x = inputs
+        if include_rescaling:
+            x = layers.Rescaling(1 / 255.0)(x)
+
+        x = layers.Conv2D(
+            64, 7, strides=2, use_bias=False, padding="same", name="conv1/conv"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name="conv1/bn"
+        )(x)
+        x = layers.Activation("relu", name="conv1/relu")(x)
+        x = layers.MaxPooling2D(3, strides=2, padding="same", name="pool1")(x)
+
+        pyramid_level_inputs = {}
+        x = apply_dense_block(x, blocks[0], name="conv2")
+        pyramid_level_inputs[2] = x.node.layer.name
+        x = apply_transition_block(x, 0.5, name="pool2")
```

Comment:
Let's expose "compression_ratio" as a top level param. Having it hang out as a const here is weird.